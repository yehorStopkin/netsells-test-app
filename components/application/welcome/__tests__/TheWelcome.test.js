const { shallowMount } = require('@vue/test-utils');
const TheWelcome = require('@/components/application/welcome/TheWelcome').default;

describe('TheWelcome', () => {
    test('component default', () => {
        const wrapper = shallowMount(TheWelcome);
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();

        const rootElement = wrapper.findAll('.the-welcome');
        expect(rootElement.length).toBe(1);

        const titleElement = wrapper.findAll('.the-welcome__title');
        expect(titleElement.length).toBe(1);

        const subtitleElement = wrapper.findAll('.the-welcome__subtitle');
        expect(subtitleElement.length).toBe(1);

        const actionElement = wrapper.findAll('.the-welcome__actions');
        expect(actionElement.length).toBe(1);
    });

    test('check click on button', () => {
        const wrapper = shallowMount(TheWelcome);

        const button = wrapper.find('*[data-test-id="app-button"]');
        expect(button.exists()).toBeTruthy();

        button.trigger('clicked');

        wrapper.vm.startEvent = jest.fn();
        expect(wrapper.vm.startEvent).toHaveBeenCalledTimes(0);
    });
});
