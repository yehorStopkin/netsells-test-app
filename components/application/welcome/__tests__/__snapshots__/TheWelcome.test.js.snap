// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheWelcome component default 1`] = `
<div
  class="the-welcome"
>
  <div
    class="the-welcome__title"
  >
    Welcome
  </div>
   
  <div
    class="the-welcome__subtitle"
  >
    Weâ€™re excited to see what you can do!
  </div>
   
  <div
    class="the-welcome__actions"
  >
    <div
      border-radius="0"
      data-test-id="app-button"
      padding-bottom="29"
      padding-top="24"
      text="BEGIN"
      width="160"
    />
  </div>
</div>
`;
